src/app/modules/group/components/pending-request/pending-request.component.ts

```diff
@@ -193,6 +197,11 @@ export class PendingRequestComponent implements OnInit, OnChanges {
 
   onSubgroupSwitch(selectedIdx: number) {
     this.includeSubgroup = this.subgroupSwitchItems[selectedIdx].includeSubgroup;
+
+    this.columns = this.columns.filter(elm => elm !== this.groupColumn);
```

Comment:
Beware that ===/!== does not work on objects in JS. Here it works only because you use your groupColumn as a constant and so the reference is the same. But actually if you move it as a clear constant,  probably that is ok for here as well.

Reply:
It's still works when I move groupColums to the class definition, can I keep !== ?

Reply:
Yes, that's even better as here, it is really the same reference as it is a constant